File: wallet/src/main/java/chatox/wallet/event/listener/UserEventsListener.java
package chatox.wallet.event.listener;

import chatox.wallet.api.response.UserResponse;
import chatox.wallet.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class UserEventsListener {
    private final UserService userService;

    @RabbitListener(queues = "wallet_service_user_created")
    public void onUserCreated(UserResponse userResponse) {
        userService.saveUser(userResponse);
    }

    @RabbitListener(queues = "wallet_service_user_updated")
    public void onUserUpdated(UserResponse userResponse) {
        userService.saveUser(userResponse);
    }
}

File: user/src/main/kotlin/chatox/user/messaging/rabbitmq/event/listener/UserEventsListener.kt
package chatox.user.messaging.rabbitmq.event.listener

import chatox.user.messaging.rabbitmq.event.AccountDeleted
import chatox.user.service.UserService
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

@Component
class UserEventsListener(private val userService: UserService) {

    @RabbitListener(queues = ["user_service_account_deleted"])
    fun onAccountDeleted(accountDeleted: AccountDeleted) = userService.deleteUsersByAccount(accountDeleted.id)
}
